node/bft/src/primary.rs

```diff
@@ -955,14 +955,53 @@ impl<N: Network> Primary<N> {
             bail!("Received a batch certificate for myself ({author})");
         }
 
+        // Retrieve the committee lookback.
+        let committee_lookback = self.ledger.get_committee_lookback_for_round(certificate_round)?;
+
+        // Ensure that the signers of the certificate reach the quorum threshold.
+        // Note that certificate.signatures() only returns the endorsing signatures, not the author's signature.
+        let mut signers: HashSet<Address<N>> =
+            certificate.signatures().map(|signature| signature.to_address()).collect();
+        signers.insert(author);
+        ensure!(
+            committee_lookback.is_quorum_threshold_reached(&signers),
+            "Certificate '{}' for round {certificate_round} does not meet quorum requirements",
+            certificate.id()
+        );
+
```

Comment:
We should already be performing this check in the subsequent call to `sync_with_certificate_from_peer` -> `insert_certificate` -> `check_certificate`.

Why not add the following `is_committee_member` checks into `check_certificate`?

Reply:
We perform the quorum signer check in check_certificate, but that's after the recursive fetching of certificates, which is the reason why the attack in #2935 works. By performing the check before the recursive fetching, we prevent that attack. Then the same check in check_certificate becomes redundant, along with essentially all of the checks done there.

Regarding the separate check that the signer are committee members, I think we could maybe safely perform it later, but it seemed cleaner to do all the necessary checks in one place (in the code added by this commit).

Reply:
That makes sense, but then it seems like we are doing these signature checks multiple times, which can get expensive. 

Additionally, for cleanliness, it might be better to create a helper method for these checks since they are done in multiple places.

Reply:
Yes, we should eliminate those other checks, but I was thinking of doing that in a separate PR?

Reply:
I added a `Storage::check_incoming_certificate()` method that could be used to check all incoming certificates from other validators. Certificates created by the current validator must already satisfy the checks by construction. So we should be able to eliminate several redundant checks, as net effect, compared to before. But we need to examine the callers a bit better, which is why I'm thinking that a separate PR to remove those redundant checks would be better.